src/ORM/FieldType/DBBoolean.php

```diff
@@ -71,16 +73,16 @@ public function scaffoldSearchField($title = null)
             0 => _t(__CLASS__ . '.NOANSWER', 'No')
         ];
 
-        return (new DropdownField($this->name, $title, $source))
+        return DropdownField::create($this->name, $title, $source)
             ->setEmptyString($anyText);
     }
 
-    public function nullValue()
+    public function nullValue(): ?int
```

Comment:
Not that I have kept the following methods nullable for all `DBField` implementations intentionally:
- `getValue()`
- `nullValue()`
- `prepValueForDB()`

This is so that subclasses can intentionally override these methods to return null. In the past I've had a business requirement for a nullable boolean and a nullable int (i.e. it's null until a content author explicitly selects an option rather than defaulting to `0` or `false`).

For that to work, these must be nullable types.